components/telemetry-operator/controllers/telemetry/logparser_controller.go

```diff
@@ -72,18 +72,26 @@ func (r *LogParserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{Requeue: shouldRetryOn(err)}, nil
 	}
 
+	var errMap = make(map[string]error)
+
 	if result.LogParserChanged {
 		if err = r.Update(ctx, &logParser); err != nil {
-			log.Error(err, "Failed updating log parser")
-			return ctrl.Result{Requeue: shouldRetryOn(err)}, err
+			errMap["Failed updating log parser"] = err
```

Comment:
You never return this error if the condition in line 83 does not match

Reply:
good catch fixed that